### ISSUE TITLE
API document for ExplainableBoosting{xxx}?

### ISSUE THREAD

**Issue by AllardJM** [interpretml/interpret#194]
I cant seem to locate an API dictionary for ExplainableBoostingRegressor or ExplainableBoostingClassifier. Does this not exist? I was looking for an understanding about what are acceptable values for feature types and other options.

**Comment by mikewlange** [comment]
Hi AllardJM, 

Not to sound like an ass, but for this kinda of question, it's better to look it up - you'll learn more. Here are a couple links I just found a second ago. But dig in, look at the source, get those hands dirty.  

https://medium.com/analytics-vidhya/explain-your-model-with-microsofts-interpretml-5daab1d693b4
https://medium.com/analytics-vidhya/model-interpretation-with-microsofts-interpret-ml-85aa0ad697ae

**Comment by AllardJM** [comment]
Yeah, well atleast you tried.

its strange for a library from a large corp not to have documentation for their API and that was the principal question - if i was missing it.

**Comment by mikewlange** [comment]
Hey AllardJM, 

Sorry - I didn't mean to be imprecise in my last comment. When you're passing data to you the Fit function (one way) would pass in returns from https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html . 

for example - pull out the arrays from your main data after munging:

`X_train, X_test, y_train, y_test = train_test_split(X_enc, y, test_size=0.20, random_state=seed)`

then for a RandomForestClassifier 

`from sklearn.ensemble import RandomForestClassifier
from sklearn.decomposition import PCA
from sklearn.pipeline import Pipeline

#Blackbox system can include preprocessing, not just a classifier!
pca = PCA()
rf = RandomForestClassifier(n_estimators=100, n_jobs=-1)
blackbox_model = Pipeline([('pca', pca), ('rf', rf)])
blackbox_model.fit(X_train, y_train)`

Please study these examples:
https://github.com/interpretml/interpret/tree/develop/examples/python/notebooks

Understand why ^ work and you're good to go.

**Comment by interpret-ml** [comment]
Hi @AllardJM -- 

Welcome back, and thanks again for your help many months ago on the calibration issue.  We don't currently have formal documentation, but we do have docstrings that detail many of the options.  You can find those at:

https://github.com/interpretml/interpret/blob/develop/python/interpret-core/interpret/glassbox/ebm/ebm.py#L1415

Hi @mikewlange -- thanks for jumping in on this issue last week.

-InterpretML

**Comment by interpret-ml** [comment]
Hey @AllardJM and @mikewlange, 

We've got official documentation now at https://interpret.ml/docs/. The EBM documentation is here: https://interpret.ml/docs/ebm.html# and includes API docs as well. Feel free to check it out and let us know what you think!

-InterpretML Team

**Comment by xiaohk** [comment]
Hey @AllardJM, thank you so much for using Interpret! I am Jay Wang, a research intern at the InterpretML team. We are developing a new visualization tool for EBM and recruiting participants for a user study (see #283 for more details).

We think you are a good fit for this paid user study! If you are interested, you can sign up with the link in #283. Let me know if you have any question. Thank you!
